lib/wellies_web/parser/open_weather.ex

```diff
@@ -0,0 +1,47 @@
+defmodule WelliesWeb.OpenWeatherParser do
+
+  def get_current_forecast(body) do
+     body
+     |> get_weather_list()
+     |> List.first()
+  end
+
+  def temperatures_grouped_by_date(body) do
+    body
+    |> get_weather_list()
+    |> clean_date_field()
+    |> group_by_date()
+  end
+
+  defp get_weather_list(body), do: Map.get(body, "list")
+
+  defp clean_date_field(element), do: Enum.map(element, fn x -> update_date(x) end)
+
+  defp group_by_date(element), do: Enum.group_by(element, fn x -> date_field(x) end)
+
+  defp update_date(element), do: Map.update!(element, "dt_txt", fn x -> get_date_from_datefield(x) end)
+
+  defp date_field(element), do: Map.get(element, "dt_txt")
```

Comment:
Try to be consistent – if you use the `map["property"]` syntax, which is  fine, stick with it. In this module you use both that one and the `Map.get` syntax, which makes me have to look twice.

On the other hand, you can use pattern matching and do the following:

```
defp date_field(%{"dt_txt" => value}), do: value
```

This has two benefits: first it's more succinct, and secondly it also enforces that the function only works with maps which contain the property "dt_txt".